vms/platformvm/state/subnet_only_validator.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"fmt"
+
+	"github.com/google/btree"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/block"
+)
+
+var _ btree.LessFunc[*SubnetOnlyValidator] = (*SubnetOnlyValidator).Less
+
+type SubnetOnlyValidator struct {
+	// ValidationID is not serialized because it is used as the key in the
+	// database, so it doesn't need to be stored in the value.
+	ValidationID ids.ID
+
+	SubnetID ids.ID     `serialize:"true"`
+	NodeID   ids.NodeID `serialize:"true"`
+
+	// PublicKey is the uncompressed BLS public key of the validator. It is
+	// guaranteed to be populated.
+	PublicKey []byte `serialize:"true"`
+
+	// Weight of this validator. It can be updated when the MinNonce is
+	// increased. If the weight is being set to 0, the validator is being
+	// removed.
+	Weight uint64 `serialize:"true"`
```

Comment:
I wonder Weight will be used in a formula to determine if SoV is an active/deactive one. I think using Weight solely could result into a misinterpration if the underlying SoV is being deleted rather than deactivating.

Reply:
Weight is not related to the status of the validator

Reply:
how/where should we put/deduce that info? as a field here (assuming we'd want to persist that info)?

Reply:
There will be different partitions in disk to include this information. This is needed because the active validator set will be iterated on startup - but the inactive validators will only live on disk (and we won't iterate over them on startup)